ffi/diplomat/src/casemap.rs

```diff
@@ -2,16 +2,50 @@
 // called LICENSE at the top level of the ICU4X source tree
 // (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
 
+use icu_casemap::titlecase::TitlecaseOptions;
+
 #[diplomat::bridge]
 pub mod ffi {
     use crate::{
         errors::ffi::ICU4XError, locale::ffi::ICU4XLocale, provider::ffi::ICU4XDataProvider,
     };
     use alloc::boxed::Box;
     use diplomat_runtime::DiplomatWriteable;
-    use icu_casemap::{CaseMapCloser, CaseMapper};
+    use icu_casemap::titlecase::{HeadAdjustment, TailCasing};
+    use icu_casemap::{CaseMapCloser, CaseMapper, TitlecaseMapper};
     use writeable::Writeable;
 
+    #[diplomat::enum_convert(HeadAdjustment, needs_wildcard)]
+    #[diplomat::rust_link(icu::casemap::titlecase::HeadAdjustment, Enum)]
+    pub enum ICU4XHeadAdjustment {
+        Adjust,
+        NoAdjust,
+    }
+
+    #[diplomat::enum_convert(TailCasing, needs_wildcard)]
+    #[diplomat::rust_link(icu::casemap::titlecase::TailCasing, Enum)]
+    pub enum ICU4XTailCasing {
+        Lowercase,
+        PreserveCase,
+    }
+
+    #[diplomat::rust_link(icu::casemap::titlecase::TitlecaseOptions, Struct)]
+    pub struct ICU4XTitlecaseOptions {
```

Comment:
Nit: Non-exhaustive options structs should have `V1` in FFI (there are not many examples of this; `ICU4XCollatorOptionsV1` is one), and functions taking them should be named `_v1`, unless you're confident that we'll never need more options here.

Reply:
And I'm pretty confident this struct won't change in the relativelt short time to 2.0, given that ICU4C/J have had these options for ages and never wanted more.

Reply:
conclusion of discussion: I'm fine with adding V1 but I want these prefixes to go away for high level FFI APIs in 2.0

Reply:
https://github.com/unicode-org/icu4x/issues/3791